lib/netns_linux.c

```diff
@@ -590,3 +592,64 @@ ns_id_t ns_get_default_id(void)
 		return default_ns->ns_id;
 	return NS_DEFAULT_INTERNAL;
 }
+
+bool netns_get_name(char *name, int len)
+{
+	char net_path[PATH_MAX];
+	int netns;
+	struct stat netst;
+	DIR *dir;
+	struct dirent *entry;
+
+	name[0] = '\0';
+
+	snprintf(net_path, sizeof(net_path), "/proc/%d/ns/net", getpid());
+	netns = open(net_path, O_RDONLY);
+	if (netns < 0) {
+		flog_warn(EC_LIB_SYSTEM_CALL,
+			  "Cannot open network namespace: %s\n",
+			  strerror(errno));
+		return false;
+	}
+	if (fstat(netns, &netst) < 0) {
+		flog_warn(EC_LIB_SYSTEM_CALL, "Stat of netns failed: %s\n",
+			  strerror(errno));
+		return false;
+	}
+	dir = opendir(NS_RUN_DIR);
+	if (!dir) {
+		/* Succeed treat a missing directory as an empty directory */
+		if (errno == ENOENT)
+			return false;
+
+		flog_warn(EC_LIB_SYSTEM_CALL,
+			  "Failed to open directory %s:%s\n", NS_RUN_DIR,
+			  strerror(errno));
+		return false;
+	}
+
+	while ((entry = readdir(dir))) {
+		char name_path[PATH_MAX];
+		struct stat st;
+
+		if (strcmp(entry->d_name, ".") == 0)
+			continue;
+		if (strcmp(entry->d_name, "..") == 0)
+			continue;
+
+		snprintf(name_path, sizeof(name_path), "%s/%s", NS_RUN_DIR,
+			 entry->d_name);
+
+		if (stat(name_path, &st) != 0)
+			continue;
+
+		if ((st.st_dev == netst.st_dev)
+		    && (st.st_ino == netst.st_ino)) {
+			strlcpy(name, entry->d_name, len);
+			closedir(dir);
+			return true;
+		}
```

Comment:
there is some code https://github.com/FRRouting/frr/blob/master/zebra/zebra_netns_notify.c#L171
that could be moved to lib/netns_linux.c, to avoid redundancy.

Reply:
That function, given a namespace name tells you if it is the default namespace.

This function returns the name of the current namespace, what's the overlap that you want to refactor?

Reply:
have a central place where to handle netns stuff:
both functions have different usage, but internally, it may call same function.